[Automation] javac compile fails for org.eclipse.jetty:jetty-servlet:9.4.0.v20161208

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26211814658
Last 300 lines of the log:
```console
 run-consecutive-tests input
========================================================================================
Coordinates: org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
Raw versions JSON: [
  "9.4.0.v20161208",
  "9.4.0.v20180619",
  "9.4.1.v20170120",
  "9.4.1.v20180619",
  "9.4.2.v20170220",
  "9.4.2.v20180619",
  "9.4.3.v20170317",
  "9.4.3.v20180619",
  "9.4.4.v20170414",
  "9.4.4.v20180619",
  "9.4.5.v20170502",
  "9.4.5.v20180619",
  "9.4.6.v20170531",
  "9.4.6.v20180619",
  "9.4.7.RC0",
  "9.4.7.v20170914",
  "9.4.7.v20180619",
  "9.4.8.v20171121",
  "9.4.8.v20180619",
  "9.4.9.v20180320",
  "9.4.10.RC0",
  "9.4.10.RC1",
  "9.4.10.v20180503",
  "9.4.11.v20180605",
  "9.4.12.RC0",
  "9.4.12.RC1",
  "9.4.12.RC2",
  "9.4.12.v20180830",
  "9.4.13.v20181111",
  "9.4.14.v20181114"
]
Parsed version count: 30
Parsed versions: 9.4.0.v20161208 9.4.0.v20180619 9.4.1.v20170120 9.4.1.v20180619 9.4.2.v20170220 9.4.2.v20180619 9.4.3.v20170317 9.4.3.v20180619 9.4.4.v20170414 9.4.4.v20180619 9.4.5.v20170502 9.4.5.v20180619 9.4.6.v20170531 9.4.6.v20180619 9.4.7.RC0 9.4.7.v20170914 9.4.7.v20180619 9.4.8.v20171121 9.4.8.v20180619 9.4.9.v20180320 9.4.10.RC0 9.4.10.RC1 9.4.10.v20180503 9.4.11.v20180605 9.4.12.RC0 9.4.12.RC1 9.4.12.RC2 9.4.12.v20180830 9.4.13.v20181111 9.4.14.v20181114
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208
========================================================================================
========================================================================================
 org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208 stage native-image run
========================================================================================
Starting stage 'native-image run' for 9.4.0.v20161208 at 2026-05-21T07:33:23Z
Command: env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED

/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                  ^
2 errors
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
          String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
          String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 4s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.eclipse.jetty:jetty-servlet:9.3.19.v20170502 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 9.4.0.v20161208 at 2026-05-21T07:33:36Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 9.4.0.v20161208 failed with exit code 1.
SOFT-FAIL[native-image run][9.4.0.v20161208][env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502]
========================================================================================
 org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208 stage javac compile
========================================================================================
Starting stage 'javac compile' for 9.4.0.v20161208 at 2026-05-21T07:33:36Z
Command: env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 947ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                  ^
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
          String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
          String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 9.4.0.v20161208 at 2026-05-21T07:33:39Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 9.4.0.v20161208 failed with exit code 1.
FAILED[javac compile][9.4.0.v20161208][env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26211814658
Last 300 lines of the log:
```console
 run-consecutive-tests input
========================================================================================
Coordinates: org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
Raw versions JSON: [
  "9.4.0.v20161208",
  "9.4.0.v20180619",
  "9.4.1.v20170120",
  "9.4.1.v20180619",
  "9.4.2.v20170220",
  "9.4.2.v20180619",
  "9.4.3.v20170317",
  "9.4.3.v20180619",
  "9.4.4.v20170414",
  "9.4.4.v20180619",
  "9.4.5.v20170502",
  "9.4.5.v20180619",
  "9.4.6.v20170531",
  "9.4.6.v20180619",
  "9.4.7.RC0",
  "9.4.7.v20170914",
  "9.4.7.v20180619",
  "9.4.8.v20171121",
  "9.4.8.v20180619",
  "9.4.9.v20180320",
  "9.4.10.RC0",
  "9.4.10.RC1",
  "9.4.10.v20180503",
  "9.4.11.v20180605",
  "9.4.12.RC0",
  "9.4.12.RC1",
  "9.4.12.RC2",
  "9.4.12.v20180830",
  "9.4.13.v20181111",
  "9.4.14.v20181114"
]
Parsed version count: 30
Parsed versions: 9.4.0.v20161208 9.4.0.v20180619 9.4.1.v20170120 9.4.1.v20180619 9.4.2.v20170220 9.4.2.v20180619 9.4.3.v20170317 9.4.3.v20180619 9.4.4.v20170414 9.4.4.v20180619 9.4.5.v20170502 9.4.5.v20180619 9.4.6.v20170531 9.4.6.v20180619 9.4.7.RC0 9.4.7.v20170914 9.4.7.v20180619 9.4.8.v20171121 9.4.8.v20180619 9.4.9.v20180320 9.4.10.RC0 9.4.10.RC1 9.4.10.v20180503 9.4.11.v20180605 9.4.12.RC0 9.4.12.RC1 9.4.12.RC2 9.4.12.v20180830 9.4.13.v20181111 9.4.14.v20181114
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208
========================================================================================
========================================================================================
 org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208 stage native-image run
========================================================================================
Starting stage 'native-image run' for 9.4.0.v20161208 at 2026-05-21T07:33:26Z
Command: env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                  ^
2 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
          String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
          String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 4s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.eclipse.jetty:jetty-servlet:9.3.19.v20170502 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 12s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 9.4.0.v20161208 at 2026-05-21T07:33:39Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 9.4.0.v20161208 failed with exit code 1.
SOFT-FAIL[native-image run][9.4.0.v20161208][env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502]
========================================================================================
 org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208 stage javac compile
========================================================================================
Starting stage 'javac compile' for 9.4.0.v20161208 at 2026-05-21T07:33:39Z
Command: env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 938ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                  ^
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
          String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
          String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 9.4.0.v20161208 at 2026-05-21T07:33:42Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 9.4.0.v20161208 failed with exit code 1.
FAILED[javac compile][9.4.0.v20161208][env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26211814658
Last 300 lines of the log:
```console
 run-consecutive-tests input
========================================================================================
Coordinates: org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
Raw versions JSON: [
  "9.4.0.v20161208",
  "9.4.0.v20180619",
  "9.4.1.v20170120",
  "9.4.1.v20180619",
  "9.4.2.v20170220",
  "9.4.2.v20180619",
  "9.4.3.v20170317",
  "9.4.3.v20180619",
  "9.4.4.v20170414",
  "9.4.4.v20180619",
  "9.4.5.v20170502",
  "9.4.5.v20180619",
  "9.4.6.v20170531",
  "9.4.6.v20180619",
  "9.4.7.RC0",
  "9.4.7.v20170914",
  "9.4.7.v20180619",
  "9.4.8.v20171121",
  "9.4.8.v20180619",
  "9.4.9.v20180320",
  "9.4.10.RC0",
  "9.4.10.RC1",
  "9.4.10.v20180503",
  "9.4.11.v20180605",
  "9.4.12.RC0",
  "9.4.12.RC1",
  "9.4.12.RC2",
  "9.4.12.v20180830",
  "9.4.13.v20181111",
  "9.4.14.v20181114"
]
Parsed version count: 30
Parsed versions: 9.4.0.v20161208 9.4.0.v20180619 9.4.1.v20170120 9.4.1.v20180619 9.4.2.v20170220 9.4.2.v20180619 9.4.3.v20170317 9.4.3.v20180619 9.4.4.v20170414 9.4.4.v20180619 9.4.5.v20170502 9.4.5.v20180619 9.4.6.v20170531 9.4.6.v20180619 9.4.7.RC0 9.4.7.v20170914 9.4.7.v20180619 9.4.8.v20171121 9.4.8.v20180619 9.4.9.v20180320 9.4.10.RC0 9.4.10.RC1 9.4.10.v20180503 9.4.11.v20180605 9.4.12.RC0 9.4.12.RC1 9.4.12.RC2 9.4.12.v20180830 9.4.13.v20181111 9.4.14.v20181114
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208
========================================================================================
========================================================================================
 org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208 stage native-image run
========================================================================================
Starting stage 'native-image run' for 9.4.0.v20161208 at 2026-05-21T07:33:24Z
Command: env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                  ^
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
          String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
          String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.eclipse.jetty:jetty-servlet:9.3.19.v20170502 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 9.4.0.v20161208 at 2026-05-21T07:33:37Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 9.4.0.v20161208 failed with exit code 1.
SOFT-FAIL[native-image run][9.4.0.v20161208][env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502]
========================================================================================
 org.eclipse.jetty:jetty-servlet:9.3.19.v20170502:9.4.0.v20161208 stage javac compile
========================================================================================
Starting stage 'javac compile' for 9.4.0.v20161208 at 2026-05-21T07:33:37Z
Command: env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 865ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                  ^
2 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:25: error: getNameOfJspClass(String) has private access in ServletHolder
          String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                      ^
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jetty/jetty-servlet/9.3.19.v20170502/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java:34: error: getPackageOfJspClass(String) has private access in ServletHolder
          String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
                                    ^
  2 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 9.4.0.v20161208 at 2026-05-21T07:33:40Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 9.4.0.v20161208 failed with exit code 1.
FAILED[javac compile][9.4.0.v20161208][env GVM_TCK_LV=9.4.0.v20161208 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.eclipse.jetty:jetty-servlet:9.3.19.v20170502]
```

--- a/metadata/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/reachability-metadata.json
+++ b/metadata/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/reachability-metadata.json
@@ -1,0 +1,113 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.server.handler.ContextHandler$Context"
+      },
+      "type" : "java.lang.StackTraceElement[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.DateCache"
+      },
+      "type" : "java.lang.String[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.ServletHolder"
+      },
+      "type" : "java.lang.reflect.Method[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.listener.ELContextCleaner"
+      },
+      "type" : "javax.el.BeanELResolver",
+      "fields" : [
+        {
+          "name" : "properties"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.Loader"
+      },
+      "type" : "javax.el.BeanELResolver"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.http.PreEncodedHttpField"
+      },
+      "type" : "org.eclipse.jetty.http.Http1FieldPreEncoder"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.ServletContextHandler"
+      },
+      "type" : "org.eclipse.jetty.security.ConstraintSecurityHandler",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.server.handler.ContextHandler$Context"
+      },
+      "type" : "org.eclipse.jetty.servlet.DefaultServlet",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.LazyList"
+      },
+      "type" : "org.eclipse.jetty.servlet.FilterHolder[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.LazyList"
+      },
+      "type" : "org.eclipse.jetty.servlet.FilterMapping[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.LazyList"
+      },
+      "type" : "org.eclipse.jetty.servlet.ListenerHolder[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.ArrayUtil"
+      },
+      "type" : "org.eclipse.jetty.servlet.ServletHolder[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.ArrayUtil"
+      },
+      "type" : "org.eclipse.jetty.servlet.ServletMapping[]"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.DateCache"
+      },
+      "type" : "sun.util.resources.cldr.CalendarData"
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.DefaultServlet"
+      },
+      "glob" : "jetty-dir.css"
+    }
+  ]
+}

--- a/metadata/org.eclipse.jetty/jetty-servlet/index.json
+++ b/metadata/org.eclipse.jetty/jetty-servlet/index.json
@@ -31,6 +31,18 @@
     ]
   },
   {
+    "metadata-version" : "9.4.0.v20161208",
+    "tested-versions" : [
+      "9.4.0.v20161208"
+    ],
+    "allowed-packages" : [
+      "org.eclipse.jetty.servlet",
+      "org.eclipse.jetty.http",
+      "org.eclipse.jetty.server.handler",
+      "org.eclipse.jetty.util"
+    ]
+  },
+  {
     "metadata-version" : "9.4.0.RC3",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/eclipse/jetty/jetty-servlet/$version$/jetty-servlet-$version$-sources.jar",
     "repository-url" : "https://github.com/jetty/jetty.project",

--- a/metadata/org.eclipse.jetty/jetty-servlet/index.json
+++ b/metadata/org.eclipse.jetty/jetty-servlet/index.json
@@ -32,6 +32,11 @@
   },
   {
     "metadata-version" : "9.4.0.v20161208",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/eclipse/jetty/jetty-servlet/$version$/jetty-servlet-$version$-sources.jar",
+    "repository-url" : "https://github.com/jetty/jetty.project",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/org/eclipse/jetty/jetty-servlet/$version$/jetty-servlet-$version$-tests.jar",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/eclipse/jetty/jetty-servlet/$version$/jetty-servlet-$version$-javadoc.jar",
+    "description" : "Jetty Servlet provides the servlet container layer for Eclipse Jetty, including servlet holders, contexts, filters, mappings, and session integration. It lets Jetty applications host and manage Java Servlet API components inside embedded or standalone Jetty servers.",
     "tested-versions" : [
       "9.4.0.v20161208"
     ],

--- a/stats/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/execution-metrics.json
+++ b/stats/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/execution-metrics.json
@@ -1,0 +1,111 @@
+{
+  "fix_javac_fail:2026-05-21": {
+    "timestamp": "2026-05-21T09:19:29.449438Z",
+    "library": "org.eclipse.jetty:jetty-servlet:9.4.0.v20161208",
+    "starting_commit": "04f1b00e18eac5c6ecf3c058675322ec426f2272",
+    "ending_commit": "878db395cd9630c1d64418ea7022fa4cf1931c02",
+    "previous_library": "org.eclipse.jetty:jetty-servlet:9.3.19.v20170502",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "9.4.0.v20161208",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 9,
+            "totalCalls": 9
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 10,
+        "totalCalls": 10
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 2503,
+          "missed": 9267,
+          "ratio": 0.212659,
+          "total": 11770
+        },
+        "line": {
+          "covered": 638,
+          "missed": 2063,
+          "ratio": 0.236209,
+          "total": 2701
+        },
+        "method": {
+          "covered": 137,
+          "missed": 308,
+          "ratio": 0.307865,
+          "total": 445
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "9.3.19.v20170502",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 9,
+            "totalCalls": 9
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 10,
+        "totalCalls": 10
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 2255,
+          "missed": 11128,
+          "ratio": 0.168497,
+          "total": 13383
+        },
+        "line": {
+          "covered": 576,
+          "missed": 2501,
+          "ratio": 0.187195,
+          "total": 3077
+        },
+        "method": {
+          "covered": 119,
+          "missed": 328,
+          "ratio": 0.266219,
+          "total": 447
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 58525,
+      "cached_input_tokens_used": 538112,
+      "output_tokens_used": 7447,
+      "iterations": 1,
+      "cost_usd": 0.4925,
+      "tested_library_loc": 638,
+      "metadata_entries": 18,
+      "test_only_metadata_entries": 2,
+      "previous_library_metadata_entries": 18,
+      "previous_library_test_only_metadata_entries": 6,
+      "code_coverage_percent": 23.62,
+      "previous_library_coverage_percent": 18.72
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org/apache/jasper/compiler/JspUtil.java",
+      "metadata_file": "metadata/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/stats.json
+++ b/stats/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "9.4.0.v20161208",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 9,
+          "totalCalls" : 9
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 10,
+      "totalCalls" : 10
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 2503,
+        "missed" : 9267,
+        "ratio" : 0.212659,
+        "total" : 11770
+      },
+      "line" : {
+        "covered" : 638,
+        "missed" : 2063,
+        "ratio" : 0.236209,
+        "total" : 2701
+      },
+      "method" : {
+        "covered" : 137,
+        "missed" : 308,
+        "ratio" : 0.307865,
+        "total" : 445
+      }
+    }
+  } ]
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/.gitignore
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/build.gradle
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/build.gradle
@@ -1,0 +1,28 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.eclipse.jetty:jetty-servlet:$libraryVersion"
+    testImplementation 'javax.el:javax.el-api:3.0.0'
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/gradle.properties
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.eclipse.jetty:jetty-servlet:9.4.0.v20161208
+library.version = 9.4.0.v20161208
+metadata.dir = org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/settings.gradle
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.eclipse.jetty.jetty-servlet_tests'

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org/apache/jasper/compiler/JspUtil.java
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org/apache/jasper/compiler/JspUtil.java
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org.apache.jasper.compiler;
+
+public final class JspUtil {
+    private JspUtil() {
+    }
+
+    public static String makeJavaIdentifier(String jsp) {
+        return "javaIdentifier:" + jsp;
+    }
+
+    public static String makeJavaPackage(String jsp) {
+        return "javaPackage:" + jsp;
+    }
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/DefaultServletTest.java
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/DefaultServletTest.java
@@ -1,0 +1,46 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_eclipse_jetty.jetty_servlet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Path;
+
+import javax.servlet.Servlet;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class DefaultServletTest {
+    @TempDir
+    Path resourceBase;
+
+    @Test
+    public void initLoadsDefaultDirectoryStylesheet() throws Exception {
+        Server server = new Server();
+        ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+        context.setContextPath("/");
+        context.setResourceBase(resourceBase.toUri().toASCIIString());
+        ServletHolder holder = context.addServlet(DefaultServlet.class, "/");
+        server.setHandler(context);
+
+        try {
+            server.start();
+
+            Servlet servlet = holder.getServlet();
+            assertThat(servlet).isInstanceOf(DefaultServlet.class);
+            assertThat(servlet.getServletConfig()).isNotNull();
+        } finally {
+            server.stop();
+            server.destroy();
+        }
+    }
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ELContextCleanerTest.java
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ELContextCleanerTest.java
@@ -1,0 +1,32 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_eclipse_jetty.jetty_servlet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import javax.el.BeanELResolver;
+
+import org.eclipse.jetty.servlet.listener.ELContextCleaner;
+import org.junit.jupiter.api.Test;
+
+public class ELContextCleanerTest {
+    @Test
+    public void contextDestroyedInspectsBeanELResolverPropertiesCache() {
+        ClassLoader originalClassLoader = Thread.currentThread().getContextClassLoader();
+        Thread.currentThread().setContextClassLoader(ELContextCleanerTest.class.getClassLoader());
+        try {
+            BeanELResolver resolver = new BeanELResolver();
+            ELContextCleaner cleaner = new ELContextCleaner();
+
+            assertThat(resolver).isNotNull();
+            assertThatCode(() -> cleaner.contextDestroyed(null)).doesNotThrowAnyException();
+        } finally {
+            Thread.currentThread().setContextClassLoader(originalClassLoader);
+        }
+    }
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/FilterHolderTest.java
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/FilterHolderTest.java
@@ -1,0 +1,81 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_eclipse_jetty.jetty_servlet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.eclipse.jetty.servlet.FilterHolder;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.junit.jupiter.api.Test;
+
+public class FilterHolderTest {
+    @Test
+    public void initializeCreatesFilterInstanceFromHeldClass() throws Exception {
+        FilterHolder holder = new FilterHolder(RecordingFilter.class);
+        holder.setName("recording");
+        holder.setInitParameter("mode", "test");
+        holder.setServletHandler(new ServletHandler());
+        holder.start();
+        try {
+            holder.initialize();
+
+            Filter filter = holder.getFilter();
+            assertThat(filter).isInstanceOf(RecordingFilter.class);
+            RecordingFilter recordingFilter = (RecordingFilter) filter;
+            assertThat(recordingFilter.getFilterName()).isEqualTo("recording");
+            assertThat(recordingFilter.getMode()).isEqualTo("test");
+            assertThat(recordingFilter.isInitialized()).isTrue();
+        } finally {
+            holder.stop();
+        }
+    }
+
+    public static class RecordingFilter implements Filter {
+        private boolean initialized;
+        private String filterName;
+        private String mode;
+
+        @Override
+        public void init(FilterConfig filterConfig) throws ServletException {
+            initialized = true;
+            filterName = filterConfig.getFilterName();
+            mode = filterConfig.getInitParameter("mode");
+        }
+
+        @Override
+        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+                throws IOException, ServletException {
+            chain.doFilter(request, response);
+        }
+
+        @Override
+        public void destroy() {
+            initialized = false;
+        }
+
+        public boolean isInitialized() {
+            return initialized;
+        }
+
+        public String getFilterName() {
+            return filterName;
+        }
+
+        public String getMode() {
+            return mode;
+        }
+    }
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ServletContextHandlerTest.java
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ServletContextHandlerTest.java
@@ -1,0 +1,26 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_eclipse_jetty.jetty_servlet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.eclipse.jetty.security.ConstraintSecurityHandler;
+import org.eclipse.jetty.security.SecurityHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.junit.jupiter.api.Test;
+
+public class ServletContextHandlerTest {
+    @Test
+    public void getSecurityHandlerCreatesDefaultSecurityHandlerWhenSecurityIsEnabled() {
+        ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SECURITY);
+
+        SecurityHandler securityHandler = context.getSecurityHandler();
+
+        assertThat(securityHandler).isInstanceOf(ConstraintSecurityHandler.class);
+        assertThat(context.getHandler()).isSameAs(securityHandler);
+    }
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java
@@ -18,22 +18,31 @@ import org.eclipse.jetty.servlet.ServletHolder;
 import org.junit.jupiter.api.Test;
 
 public class ServletHolderTest {
-    @Test
-    public void getNameOfJspClassUsesAvailableJasperUtility() {
-        ServletHolder holder = new ServletHolder();
-
-        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
-
-        assertThat(jspClassName).isEqualTo("javaIdentifier:monthly-report.jsp");
-    }
+    private static final String JSP_PATH = "/WEB-INF/pages/monthly-report.jsp";
+    private static final String PRECOMPILED_JSP_NAME = "org.apache.jsp.javaPackage:/WEB-INF/pages"
+            + ".javaIdentifier:monthly-report.jsp";
 
     @Test
-    public void getPackageOfJspClassUsesAvailableJasperUtility() {
+    public void startMapsForcedJspToPrecompiledServletClassUsingAvailableJasperUtility() throws Exception {
+        ServletHandler handler = new ServletHandler();
+        handler.setEnsureDefaultServlet(false);
+        ServletHolder precompiledJsp = new ServletHolder(PRECOMPILED_JSP_NAME, CountingServlet.class);
         ServletHolder holder = new ServletHolder();
+        holder.setName("monthly-report");
+        holder.setForcedPath(JSP_PATH);
 
-        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
+        handler.addServlet(precompiledJsp);
+        handler.addServlet(holder);
 
-        assertThat(jspPackage).isEqualTo("javaPackage:/WEB-INF/pages");
+        try {
+            handler.start();
+
+            assertThat(holder.getClassName()).isEqualTo(CountingServlet.class.getName());
+        } finally {
+            if (handler.isRunning()) {
+                handler.stop();
+            }
+        }
     }
 
     @Test

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/java/org_eclipse_jetty/jetty_servlet/ServletHolderTest.java
@@ -1,0 +1,66 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_eclipse_jetty.jetty_servlet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.junit.jupiter.api.Test;
+
+public class ServletHolderTest {
+    @Test
+    public void getNameOfJspClassUsesAvailableJasperUtility() {
+        ServletHolder holder = new ServletHolder();
+
+        String jspClassName = holder.getNameOfJspClass("/WEB-INF/pages/monthly-report.jsp");
+
+        assertThat(jspClassName).isEqualTo("javaIdentifier:monthly-report.jsp");
+    }
+
+    @Test
+    public void getPackageOfJspClassUsesAvailableJasperUtility() {
+        ServletHolder holder = new ServletHolder();
+
+        String jspPackage = holder.getPackageOfJspClass("/WEB-INF/pages/monthly-report.jsp");
+
+        assertThat(jspPackage).isEqualTo("javaPackage:/WEB-INF/pages");
+    }
+
+    @Test
+    public void getServletCreatesServletInstanceFromHeldClass() throws ServletException {
+        ServletHolder holder = new ServletHolder("counting", CountingServlet.class);
+        holder.setServletHandler(new ServletHandler());
+
+        Servlet servlet = holder.getServlet();
+
+        assertThat(servlet).isInstanceOf(CountingServlet.class);
+        assertThat(((CountingServlet) servlet).getServletConfig()).isNotNull();
+    }
+
+    public static class CountingServlet extends HttpServlet {
+        private static final long serialVersionUID = 1L;
+
+        private ServletConfig servletConfig;
+
+        @Override
+        public void init(ServletConfig config) throws ServletException {
+            super.init(config);
+            servletConfig = config;
+        }
+
+        @Override
+        public ServletConfig getServletConfig() {
+            return servletConfig;
+        }
+    }
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -2,32 +2,6 @@
   "reflection" : [
     {
       "condition" : {
-        "typeReached" : "org.eclipse.jetty.servlet.ServletHolder"
-      },
-      "type" : "org.apache.jasper.compiler.JspUtil",
-      "methods" : [
-        {
-          "name" : "makeJavaIdentifier",
-          "parameterTypes" : [
-            "java.lang.String"
-          ]
-        },
-        {
-          "name" : "makeJavaPackage",
-          "parameterTypes" : [
-            "java.lang.String"
-          ]
-        }
-      ]
-    },
-    {
-      "condition" : {
-        "typeReached" : "org.eclipse.jetty.util.Loader"
-      },
-      "type" : "org.apache.jasper.compiler.JspUtil"
-    },
-    {
-      "condition" : {
         "typeReached" : "org.eclipse.jetty.servlet.FilterHolder"
       },
       "type" : "org_eclipse_jetty.jetty_servlet.FilterHolderTest$RecordingFilter"

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -2,6 +2,26 @@
   "reflection" : [
     {
       "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.ServletHolder"
+      },
+      "type" : "org.apache.jasper.compiler.JspUtil",
+      "methods" : [
+        {
+          "name" : "makeJavaIdentifier",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name" : "makeJavaPackage",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
         "typeReached" : "org.eclipse.jetty.servlet.FilterHolder"
       },
       "type" : "org_eclipse_jetty.jetty_servlet.FilterHolderTest$RecordingFilter"

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,42 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.ServletHolder"
+      },
+      "type" : "org.apache.jasper.compiler.JspUtil",
+      "methods" : [
+        {
+          "name" : "makeJavaIdentifier",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name" : "makeJavaPackage",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.util.Loader"
+      },
+      "type" : "org.apache.jasper.compiler.JspUtil"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.FilterHolder"
+      },
+      "type" : "org_eclipse_jetty.jetty_servlet.FilterHolderTest$RecordingFilter"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.eclipse.jetty.servlet.ServletHolder"
+      },
+      "type" : "org_eclipse_jetty.jetty_servlet.ServletHolderTest$CountingServlet"
+    }
+  ]
+}

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/user-code-filter.json
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/user-code-filter.json
@@ -7,6 +7,15 @@
       "includeClasses" : "org.eclipse.jetty.servlet.**"
     },
     {
+      "includeClasses" : "org.eclipse.jetty.http.**"
+    },
+    {
+      "includeClasses" : "org.eclipse.jetty.server.handler.**"
+    },
+    {
+      "includeClasses" : "org.eclipse.jetty.util.**"
+    },
+    {
       "includeClasses" : "org.apache.jasper.compiler.**"
     },
     {

--- a/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/user-code-filter.json
+++ b/tests/src/org.eclipse.jetty/jetty-servlet/9.4.0.v20161208/user-code-filter.json
@@ -1,0 +1,16 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.eclipse.jetty.servlet.**"
+    },
+    {
+      "includeClasses" : "org.apache.jasper.compiler.**"
+    },
+    {
+      "includeClasses" : "org_eclipse_jetty.jetty_servlet.**"
+    }
+  ]
+}

